design/ceph/multus-network.md

```diff
@@ -0,0 +1,110 @@
+Also, this removes the need of privileged containers (required for `HostNetworking`).
+
+## Proposed CRD changed
+
+We already have a `network` CRD property, which looks like:
+
+```yaml
+network:
+  provider:
+  selectors:
+```
+
+We will expand the `selectors` with the following two hardcoded keys:
+
+```yaml
+selectors:
+  public:
+  cluster:
+```
+
+Each selector represents a [`NetworkAttachmentDefinition`](https://github.com/intel/multus-cni/blob/master/doc/quickstart.md#storing-a-configuration-as-a-custom-resource) object in Multus.
+At least, one must be provided and by default they will represent:
+
+- `public`: data daemon public network (binds to `public_network` Ceph option).
+- `cluster`: data daemon cluster network (binds to `cluster_network` Ceph option).
+
+If only `public` is set then `cluster` will take the value of `public`.
+
+## Multus supported configuration
+
+### Interface type
+
+As part of the CNI spec, Multus supports several [interface types](https://github.com/containernetworking/plugins#main-interface-creating).
+Rook will naturally support any of them as they don't fundamentally change the working behavior.
+
+### IPAM type
+
+This is where things get more complex.
+Currently there are [three different IPAM](https://github.com/containernetworking/plugins#ipam-ip-address-allocation) solutions available.
+
+As part of my research I have found that the following IPAM types are not good candidates:
+
+- host-local: Maintains a local database of allocated IPs, this only works on a per host basis, so not suitable for a distributed environment since we will end up with IP collision
+To fix this, the [whereabouts project](https://github.com/dougbtv/whereabouts) looks promising but is not officially supported.
+- static: Allocate a static IPv4/IPv6 addresses to container and it's useful in debugging purpose.
+This cannot at scale because this means we will have to allocate IPs for **all** the daemon so it's not scalable.
+
+## Monitors and OSDs implementation challenges
+
+The Ceph monitors only need access to the public network.
+The OSDs needs access to both public and cluster networks.
+
+Monitors requirements so far are the following:
+
+- Predictable IP addresses
+- Keep the same IP address for the life time of the pod (IP should survive a restart)
+
+### First proposal: IPAM type 'DHCP'
+
+In this scenario, a DHCP server will distribute an IP address to a pod using a given range.
+
+Pros:
+
+- Pods will get a dedicated IP on a physical network interface
+- No changes required in Ceph, Rook will detect the CIDR via the `NetworkAttachmentDefinition` then populate Ceph's flag `public_network` and `cluster_network`
+
+Cons:
+
+- IP allocation not predictable, we don't know it until the pod is up and running.
+So the detection must append inside the monitor container is running, similarly to what the OSD code does today.
+- Requires drastic changes in the monitor bootstrap code
```

Comment:
How can this work at all for mons? If the mon IP changes, the core identity of the mon changes. It seems like we have to change the mons to allow for changing IP addresses, which I didn't think was an option.

Reply:
Well, I think there is an aggressive option where failover mons more frequently...